DB/Item.lua

```diff
@@ -150,19 +150,118 @@ function Item:IsMount(entry)
 	return entry.type == CONSTANTS.ITEM_TYPES.MOUNT
 end
 
-function Item:HasZone(entry)
-	if not entry.coords then
+function Item:IsCollectionItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.COLLECTION
+end
+
+function Item:IsValidCollectionItem(entry)
+	return entry.collectedItemId ~= nil
+end
+
+function Item:IsArchaeologyItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.ARCH
+end
+
+function Item:IsValidArchaeologyItem(entry)
+	if not entry.raceId then
+		Rarity:Print("Found archaeology item without raceId property")
+		return false
+	else
+		for _, constantRaceIDs in pairs(CONSTANTS.ARCHAEOLOGY_RACES) do
+			if constantRaceIDs == entry.raceId then
+				return true
+			end
+		end
+
+		Rarity:Print("Found archaeology item with faulty raceId")
 		return false
 	end
+end
 
-	-- It has value in 'coords', make sure it contains a mapID.
-	for _, waypointData in pairs(entry.coords) do
-		if waypointData.m then
-			return true
+function Item:HasUseMethod(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.USE
+end
+
+function Item:IsValidUseMethodItem(entry)
+	return entry.items ~= nil
+end
+
+function Item:IsCovenantSpecificItem(entry)
+	return entry.requiresCovenant == true
+end
+
+function Item:IsValidCovenantItem(entry)
+	if not entry.requiredCovenantID then
+		Rarity:Print("Found covenant item without a covenantID property")
+		return false
+	else
+		for _, constantIDs in pairs(CONSTANTS.COVENANT_IDS) do
+			if constantIDs == entry.requiredCovenantID then
+				return true
+			end
 		end
+
+		Rarity:Print("Found covenant item with faulty covenantID")
+		return false
 	end
+end
+
+function Item:IsZoneItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.ZONE
+end
 
-	return false
+function Item:IsValidZoneItem(entry)
+	return entry.zones ~= nil
+end
+
+function Item:IsFishingItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.FISHING
+end
+
+function Item:IsValidFishingItem(entry)
+	if not Item:IsValidZoneItem(entry) then
+		return false
+	else
+		return entry.requiresPool ~= nil
+	end
+end
+
+function Item:HasNPCMethod(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.NPC
+end
+
+function Item:HasBossMethod(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.BOSS
+end
+
+function Item:IsValidNPCItem(entry)
+	return entry.npcs ~= nil
+end
+
+function Item:HasWaypointData(entry)
+	return entry.coords ~= nil
+end
+
+function Item:HasValidWaypoints(entry)
+	for _, waypoint in pairs(entry.coords) do
+		if not IsValidWaypoint(waypoint) then
+			return false
+		end
+	end
+	return true
+end
+
+function IsValidWaypoint(waypointData)
+	if type(waypointData) == "table" then
+		if not waypointData.m then
+			Rarity:Print("Missing mapID")
+			return false
+		elseif (waypointData.y and not waypointData.x) or (waypointData.x and not waypointData.y) then
+			Rarity:Print("X/Y-Coordinates are inconsistent")
```

Comment:
Same here; "inconsistent" doesn't really mean much so I'd probably rephrase this to something like "Waypoint data has incomplete map coordinates". Could also use "invalid" or "partial" here, but that doesn't quite sound as clear (to me, anyway). What do you think?

Reply:
Rephrased this in 7bb53d56902e48e69116e48dbe8cca91c4fe8179.